corehq/apps/users/models.py

```diff
@@ -329,11 +329,24 @@ def ids_of_assigned_users(self):
 
         return assigned_user_ids
 
+    @classmethod
+    def get_preset_permission_by_name(cls, name):
+        matches = {k for k, v in PERMISSIONS_PRESETS.iteritems() if v['name'] == name}
+        return matches.pop() if matches else None
+
+    @classmethod
+    def preset_and_domain_role_names(cls, domain_name):
+        return cls.preset_permissions_names().union(set([role.name for role in cls.by_domain(domain_name)]))
+
+    @classmethod
+    def preset_permissions_names(cls):
+        return {details['name'] for role, details in PERMISSIONS_PRESETS.iteritems()}
+
 PERMISSIONS_PRESETS = {
     'edit-apps': {'name': 'App Editor', 'permissions': Permissions(edit_apps=True, view_reports=True)},
     'field-implementer': {'name': 'Field Implementer', 'permissions': Permissions(edit_commcare_users=True, view_reports=True)},
     'read-only': {'name': 'Read Only', 'permissions': Permissions(view_reports=True)},
-    'no-permissions': {'name': 'Read Only', 'permissions': Permissions(view_reports=True)},
+    'no-permissions': {'name': 'Read Only(No Permissions)', 'permissions': Permissions(view_reports=True)},
```

Comment:
what's the deal with this / where does it show up? (it seems like if we're changing the name there should be a space between "Only" and the "(")